blockly/src/server/Server.java

```diff
@@ -1328,28 +1331,47 @@ private void waitDelta() {
   }
 
   /**
-   * Check if the hero is near a wall in a specific direction.
+   * Check if the next tile in the given direction is an instance from the given class.
    *
-   * <p>This method checks if the hero is near a wall in the specified direction. It retrieves the
-   * hero's current coordinates and calculates the target coordinates based on the direction. It
-   * then checks if the target tile is accessible. If the hero or the target tile is null or not
-   * accessible, it returns true, indicating that the hero is near a wall.
-   *
-   * @param direction Direction in which the hero will be moved.
-   * @return Returns true if the hero is null or the target tile is not accessible. Otherwise,
+   * @param tileClass Tile-Class to check for.
+   * @param direction Direction to check
+   * @return Returns true if the hero is null or the target tile is from the given class. Otherwise,
    *     returns false.
    */
-  public boolean isNearWall(final Direction direction) {
+  public boolean isNearTile(Class<? extends Tile> tileClass, final Direction direction) {
     Coordinate heroCoords = EntityUtils.getHeroCoordinate();
     if (heroCoords == null) {
       return true;
     }
     Coordinate targetCoords = heroCoords.add(new Coordinate(direction.x(), direction.y()));
     Tile targetTile = Game.tileAT(targetCoords);
     if (targetTile == null) {
+      return false;
+    }
+    return tileClass.isInstance(targetTile);
+  }
+
+  /**
+   * Check if on the next tile in the given direction an entity with the given component exist.
+   *
+   * @param componentClass Component-Class to check for.
+   * @param direction Direction to check
+   * @return Returns true if the hero is null or a entity with the given component was detected.
+   *     Otherwise, returns false.
+   */
+  public boolean isNearComponent(
+      Class<? extends Component> componentClass, final Direction direction) {
+    Coordinate heroCoords = EntityUtils.getHeroCoordinate();
+    if (heroCoords == null) {
       return true;
     }
-    return !targetTile.isAccessible();
+    Coordinate targetCoords = heroCoords.add(new Coordinate(direction.x(), direction.y()));
```

Comment:
Mittlerweile haben wir eine `.toPoint()`
```suggestion
    Coordinate targetCoords = heroCoords.add(direction.toPoint().toCoordinate());
```